app/src/main/java/com/example/dsublett/monsterfetch/activities/ClassDetail.kt

```diff
@@ -16,18 +17,19 @@ class ClassDetail : DetailActivity("classFavorites") {
         setContentView(R.layout.class_detail)
 
         this.classDetailView.visibility = View.INVISIBLE
-        this.initSharedPreferences()
         this.itemIndex =
             UrlParse.getIndex(this.intent.getParcelableExtra<ResponseItem>("responseItem").url)
-
+        this.initSharedPreferences()
         ServiceProxy.dndService.getClass(this.itemIndex, this::buildUI, this::logFailure)
     }
 
-    private fun buildUI(details: ClassResponse?) {
+    override fun buildUI(details: ItemResponse?) {
```

Comment:
Since this is not fully implemented, i would go ahead and revert it back to `private` just so you don't get confused if later on ur looking at this code and attempting to decipher why you needed to override this method.